### PR TITLE
#4000: lazy load sidebar content/modules

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Part of #4000
- Lazy load sidebar content/modules as-needed to reduce de-facto memory footprint
- Puts the following sidebar content behind lazy load/suspense:
  - [x] Mod Launcher 
  - [x] Activate Mod panels
- Don't mount tab content in sidebar until they become active for first time
  - Mod Launcher: important because it hydrates modDefinitions slice and makes call to marketplace
  - Persistent Mod Panels (i.e., Sidebar Starter Brick): important because panels may re-import bootstrap, etc.

## Discussion

- Previously this PR also split out the Extension Console screens. See future work

## Remaining Work
 
- [x] Fix broken test: "can open a closed panel when mod launcher is available"
- [x] Fix background error when using close tab (seems related to default action on anchor?): https://www.loom.com/share/0b56e5890e1043de86aec7bf4efe736d?sid=bd28dbcb-995e-4a0c-ace9-d86b856fde70 -- I added a preventDefault on the close action
- [ ] Fix broken test on CI (is working locally): "Tabs › Mod Launcher › renders with mod launcher"

## Future Work

- The sidebar loads a lot of content into memory (e.g., complete Mod Definitions for all mods available to the user) that doesn't end up getting used
- Module split the Extension Console Screens. I was getting aa Chrome errors in the background for "Uncaught (in promise) Error: Cannot access contents of url "chrome-extension://mpjjildhmpddojocokjkgmlkkkfjnepo/options.html#/". Extension manifest must request permission to access this host." error. (This might be due to an automatic redirect?)

## Checklist

- [x] Add tests: N/A
- [x] New files added to `src/tsconfig.strictNullChecks.json` (if possible): N/A
- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

**Comment by twschiller** [comment]
> Look into Chrome errors in background for "Uncaught (in promise) Error: Cannot access contents of url "chrome-extension://mpjjildhmpddojocokjkgmlkkkfjnepo/options.html#/". Extension manifest must request permission to access this host."

@fregante any ideas here? The screens are loading locally for me

File: src/extensionConsole/App.tsx [hunk 1]
```diff
@@ -56,6 +49,64 @@ import ReduxPersistenceContext, {
...
+      /* webpackChunkName: "WorkshopPage" */
+      "@/extensionConsole/pages/brickEditor/EditPage"
+    )
+);
+
+const WorkshopPage = lazy(
+  async () =>
+    import(
+      /* webpackChunkName: "WorkshopPage" */
+      "./pages/workshop/WorkshopPage"
```
Comment (fregante) [inline src/extensionConsole/App.tsx:98]:
```suggestion
      "@/extensionConsole/pages/workshop/WorkshopPage"
```

**Review by fregante — COMMENTED** [review]
> * Chrome errors in background for "Uncaught (in promise) Error: Cannot access contents of url "chrome-extension://mpjjildhmpddojocokjkgmlkkkfjnepo/options.html#/". Extension manifest must request permission to access this host."


I'm not seeing that. What's the repro?

Screenshot 17

**Comment by twschiller** [comment]
> I'm not seeing that. What's the repro?

I just dropped that code to get the more important content merge. The error was appearing in the console when I had dynamic imports in `App.tsx` for screens in the Extension Console https://github.com/pixiebrix/pixiebrix-extension/pull/6780#discussion_r1378478053

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**